### ISSUE TITLE
Investigate running AMP tests on Safari Technology Preview using MacOS VMs

### ISSUE THREAD

**Issue by rsimha** [ampproject/amphtml#27044]
We'd like to be able to run (a subset of) AMP tests on [Safari Technology Preview](https://developer.apple.com/safari/technology-preview/). Some things to investigate:

- Compare Travis's [MacOS VMs](https://docs.travis-ci.com/user/reference/osx/#using-macos) against [Github Actions](https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners). (#24088 tried this a few months ago.)
- We should aim to run unit, integration, and e2e tests. Do they run fine and pass?
- What is the start-up time of the VMs, and running time of the tests?
- Based on the above answers, should the tests be part of PR builds, push builds, or run when a new nightly branch is cut?

**Comment by rcebulko** [comment]
According to the [Usage Limits](https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits) section of the GitHub Actions docs, all GitHub plans except for Enterprise cap concurrent macOS VMs at 5 instances. That seems like a potential challenge to use Actions for our CI, since at peak contributing hours we often have at last that many PR and/or push builds running, and that concurrency limit could slow things down.

I'm experimenting with MacOS Travis builds right now to compare. Depending on how Travis handles it, we may have to stick with only running on `master` push builds, so as not to block developers with long Travis build delays on e2e

**Comment by rcebulko** [comment]
In #27070 I tried running e2e tests on MacOS on Safari (not TP) and [as you can see](https://travis-ci.org/ampproject/amphtml/jobs/657853842?utm_medium=notification&utm_source=github_status), it took somewhere around 4x as long to run the tests, and around 6x as long to run the job overall compared to Travis Linux VMs. There were also some failing tests on Safari and other browsers, but I can't tell if these are flakes/already failing.

TL;DR - Safari/MacOS testing is _really slow_.

It would be reasonable to run all test suites on Safari on nightly release builds. Trying to incorporate them into PR builds or even push builds would add a lot of wall clock time to the builds and consume a lot of additional Travis VMs in the process (not sure how big a concern this is). Using Travis to do so would probably be preferable to GitHub Actions, since it would mostly be configuration changes, as opposed to GitHub Actions which would involve introducing a whole new workflow/system to the repo

**Comment by rsimha** [comment]
_[Infra triage]_ Putting this on the back-burner in light of the comments above. This can be revisited if / when are able to move cross-browser testing to CircleCI.
